fix #6 Reset Button

mmorihiro opened the pull request:
#6

mmorihiro committed:
Merge pull request #3 from MMorihiro/LifeGame-2

Displaying a button (Fix #2)

mmorihiro committed:
Merge pull request #10 from MMorihiro/simpleButton

#1 Simple button

mmorihiro committed:
#11 Resolve conflict

mmorihiro committed:
#5 Start Button

mmorihiro committed:
#4 Add a table for buttons

mmorihiro committed:
#4 Stop Button

mmorihiro committed:
#6 Reset Button

mmorihiro committed:
#6 Reset Button

mmorihiro commented on core/src/main/com/github/mmorihiro/core/GameState.kt:
```diff
@@ -1,55 +1,24 @@
 
-
-class GameState {
+internal class GameState {
```
わざわざ`public`にする必要もないので公開範囲を限定。

mmorihiro commented on core/src/main/com/github/mmorihiro/core/SetUpUILook.kt:
```diff
@@ -0,0 +1,45 @@
+import com.badlogic.gdx.scenes.scene2d.ui.TextButton
+
+
+internal class SetUpUILook(generation: Int) {
```
`GameState`の処理が多くなってきたので、UIの見た目に関する部分を抽出。

mmorihiro commented on core/src/main/com/github/mmorihiro/core/GameState.kt:
```diff
@@ -58,12 +27,26 @@ class GameState {
         }
     }
+
+    fun stage() = setup.stage
```
stageを`SetUpUILook`に移動したので、stageを`LifeGame`が引き続き取得出来るようにゲッターメソッドを追加。

mmorihiro commented on core/src/main/com/github/mmorihiro/core/GameState.kt:
```diff
@@ -1,55 +1,24 @@
     private fun addActionToButtons() {
+        val startButton = setup.startButton
+        val stopButton = setup.stopButton
+        val resetButton = setup.resetButton
```
いちいち`setup.somebutton`と書くのは面倒なので、一度メソッド内で参照を取得してしまう。

forestone23 commented on core/src/main/com/github/mmorihiro/core/GameState.kt:
```diff
@@ -58,12 +27,26 @@ class GameState {
+                setLabelText(generation)
+                canIncrement = false
+            }
+            stopButton.isPressed
```
ここ`resetButton.isPressed`だよね

mmorihiro replied:
そこね。コピペの際に修正し忘れてた。

mmorihiro committed:
#11 Change `stage()` method to value

mmorihiro committed:
#11 Dispose skin

mmorihiro committed:
fix #11 Increment generation

Merge pull request #12 from increment-generation

mmorihiro committed:
fix #5 Start Button

Merge pull request #13 from MMorihiro/5_Start_Button

mmorihiro committed:
fix #4 stop button

Merge pull request #23 from MMorihiro/4_Stop_Button

mmorihiro committed:
#15 Merge remote-tracking branch 'remotes/origin/develop' into 6_Reset_Button

# Conflicts:
#	core/src/main/com/github/mmorihiro/core/GameState.kt
#	core/src/main/com/github/mmorihiro/core/LifeGame.kt

Commit:
#15 Merge remote-tracking branch 'remotes/origin/develop' into 6_Reset_Button

# Conflicts:
#	core/src/main/com/github/mmorihiro/core/GameState.kt
#	core/src/main/com/github/mmorihiro/core/LifeGame.kt

Commit:
#15 Fix button kind in addListener

mmorihiro committed:
#15 Fix button kind in addListener

Commit:
Merge remote-tracking branch 'origin/6_Reset_Button' into 6_Reset_Button

mmorihiro committed:
Merge remote-tracking branch 'origin/6_Reset_Button' into 6_Reset_Button

mmorihiro committed:
#15 Resolve conflict

mmorihiro committed:
Merge remote-tracking branch 'origin/6_Reset_Button' into 6_Reset_Button

# Conflicts:
#	core/src/main/com/github/mmorihiro/core/GameState.kt
#	core/src/main/com/github/mmorihiro/core/LifeGame.kt
#	core/src/main/com/github/mmorihiro/core/SetUpUILook.kt

mmorihiro committed:
#15 Resolve conflict

mmorihiro committed:
Merge remote-tracking branch 'origin/6_Reset_Button' into 6_Reset_Button
